ansible_collections/arista/avd/plugins/plugin_utils/schema/avdschemaresolver.py

```diff
@@ -7,99 +7,111 @@
 
-    In place update of supplied child_schema
-    """
-    scope, ref_schema = validator.resolver.resolve(ref)
-    ref_schema = deepcopy(ref_schema)
-    child_schema.pop("$ref", None)
-    merge(child_schema, ref_schema, same_key_strategy="use_existing")
-    # Resolve new refs inherited from the first ref.
-    if "$ref" in child_schema:
-        _ref_on_child(validator, child_schema["$ref"], child_schema)
+            self.resolve(resolved_schema["keys"][key])
 
+    def _dynamic_keys(self, resolved_schema: dict):
+        for key in resolved_schema["dynamic_keys"]:
+            # Resolve the child schema
+            # Repeat in case new refs inherited from the first ref.
+            while "$ref" in resolved_schema["dynamic_keys"][key]:
+                self._ref_on_child(resolved_schema["dynamic_keys"][key])
 
-class AvdSchemaResolver:
-    def __new__(cls, schema, store):
-        """
-        AvdSchemaResolver is used to resolve $ref in AVD Schemas.
+            self.resolve(resolved_schema["dynamic_keys"][key])
+
+    def _items(self, resolved_schema: dict):
+        # Resolve the child schema
+        # Repeat in case new refs inherited from the first ref.
+        while "$ref" in resolved_schema["items"]:
+            self._ref_on_child(resolved_schema["items"])
 
-        It is used to generate full documentation covering all nested schemas.
+        self.resolve(resolved_schema["items"])
 
-        Since we return generators, we cannot also return the resolved schema.
-        Instead we use the "instance" in jsonschema - the variable normally holding
-        the data to be validated - called "resolved_schema" above.
-        The "resolved_schema" must contain a copy of the original schema, and then
-        the $ref resolver will merge in the resolved schema and do in-place update.
+    def _ref_on_child(self, resolved_schema: dict):
         """
-        if JSONSCHEMA_IMPORT_ERROR:
-            raise AristaAvdError('Python library "jsonschema" must be installed to use this plugin') from JSONSCHEMA_IMPORT_ERROR
-
-        ValidatorClass = jsonschema.validators.create(
-            meta_schema=store["avd_meta_schema"],
-            validators={
-                "items": _items,
-                "keys": _keys,
-                "dynamic_keys": _dynamic_keys,
-            },
-        )
+        This function resolves the $ref referenced schema,
+        then merges with any schema defined at the same level
 
-        return ValidatorClass(schema, resolver=create_refresolver(schema, store))
+        In place update of supplied resolved_schema
+        """
+        resolved = self.resolver.lookup(resolved_schema["$ref"])
+        ref_schema = deepcopy(resolved.contents)
+        resolved_schema.pop("$ref")
+        merge(resolved_schema, ref_schema, same_key_strategy="use_existing", list_merge="replace")
+
+    def create_resolver(self, store: dict, base_uri=""):
+        registry = self.create_registry(store)
```

Comment:
if it is a static method should we not be using `<CLASS>.create_registry`? It works with self from what I can read so no real opinion here - just questioning how we want to do this